### PR TITLE
fix: Show unrecognized topic and "Unknown Topic (...)"

### PR DESCRIPTION
**Pull request by jasonz-dfinity**
# Motivation

Currently, if a new topic is added in NNS Governance, we need to follow the following steps:

1. Define new topics in NNS Governance, while disallowing to set followees on them
2. Add new topics to ic-js and NNS Dapp, so that following on them can be shown, but cannot be set
3. Update NNS Governance to allow following to be set
4. Update NNS Dapp to allow following to be set.

After this change, the procedure will be the following:

1. Define new topics in NNS Governance, while followees CAN be set on them
2. Add new topics to ic-js and NNS Dapp, so that following can be shown and set

# Changes

Show unrecognized topic as "Unknown Topic ($topicId)"

# Tests

unit tests and manual test

Screenshot 2024-07-18 at 2 35 46 PM

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

File: frontend/src/tests/lib/utils/neuron.utils.spec.ts [hunk 1]
```diff
@@ -1056,6 +1056,10 @@ describe("neuron-utils", () => {
               topic: Topic.Governance,
               followees: [0n, 1n, 2n],
             },
+            {
+              topic: 100 as Topic,
+              followees: [3n, 4n],
+            },
```
Comment (dskloetd) [inline frontend/src/tests/lib/utils/neuron.utils.spec.ts:1062]: Instead of changing an existing test, can you add a separate test specifically to add the skipping?

File: frontend/src/lib/utils/neuron.utils.ts [hunk 2]
```diff
@@ -486,6 +486,9 @@ export const followeesNeurons = (neuron: NeuronInfo): FolloweesNeuron[] => {
     result.find(({ neuronId: id }) => id === neuronId);
 
   for (const { followees, topic } of neuron.fullNeuron.followees) {
+    if (!(topic in Topic)) {
```
Comment (dskloetd) [inline frontend/src/lib/utils/neuron.utils.ts:489]: I'm wondering if it wouldn't be better to render the unknown topics as "Unknown topic" instead of hiding them. What do you think?

Comment (dskloetd) [inline frontend/src/lib/utils/neuron.utils.ts:489]: Can you add a comment explaining what this is for? I think it might be confusing outside the context of this PR.

File: frontend/src/tests/lib/utils/neuron.utils.spec.ts (hunk 1 above)
Comment (jasonz-dfinity) [reply frontend/src/tests/lib/utils/neuron.utils.spec.ts]: N/A as a different approach is used.

File: frontend/src/lib/utils/neuron.utils.ts (hunk 2 above)
Comment (jasonz-dfinity) [reply frontend/src/lib/utils/neuron.utils.ts]: N/A as a different approach is used.

Comment (jasonz-dfinity) [reply frontend/src/lib/utils/neuron.utils.ts]: Good idea. I tried it. Could you take another look?

File: CHANGELOG-Nns-Dapp-unreleased.md
Comment (dskloetd) [inline CHANGELOG-Nns-Dapp-unreleased.md:1]: Please don't change the formatting of the file. This is not related to the PR.

File: frontend/src/tests/lib/utils/neuron.utils.spec.ts [hunk 3]
```diff
@@ -2782,6 +2782,9 @@ describe("neuron-utils", () => {
       expect(getTopicTitle({ topic: Topic.SubnetRental, i18n: en })).toBe(
         "Subnet Rental"
       );
+      expect(getTopicTitle({ topic: 1000 as Topic, i18n: en })).toBe(
```
Comment (dskloetd) [inline frontend/src/tests/lib/utils/neuron.utils.spec.ts:2785]: How about putting this in a separate test?
```
it("should render unknown topics", ...)
```

**Review by dskloetd — APPROVED** [review]
Thanks!
